src/GCodes/GCodes2.cpp

```diff
@@ -3921,20 +3923,52 @@ bool GCodes::HandleMcode(GCodeBuffer& gb, const StringRef& reply) THROWS(GCodeEx
 
 											IPAddress ip;
 
-											if (protocol == MqttProtocol)
+#if SUPPORT_MQTT
+											if (interface < reprap.GetNetwork().GetNumNetworkInterfaces() && protocol == MqttProtocol)
```

Comment:
This code is getting too long to be included in GCodes2.cpp. I suggest you move all of the code after "if (CheckNetworkCommandAllowed(gb, reply, result)" into a new function in Network.cpp, just as we do for M587.